subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/FileZipInput.java

```diff
@@ -29,10 +32,29 @@
 
 class FileZipInput implements ZipInput {
 
+    public static ZipInput create(File file) {
```

Comment:
it's a bit weird that we have a factory that may or may not return itself.
Would it make sense to externalize the factory?